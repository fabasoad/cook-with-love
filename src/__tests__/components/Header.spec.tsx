/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Header from '../../components/Header'
import { useTranslation } from '../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render Header correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(3)
  new Array<string>(
    'cook-with-love-header-occupation',
    'cook-with-love-header-about-me',
    'cook-with-love-header-portfolio'
  ).forEach((n, i) => expect(tMock).toHaveBeenNthCalledWith(i + 1, n))
  expect(
    wrapper.find('Trans').childAt(0).text()
  ).toBe('cook-with-love-header-title')
})
