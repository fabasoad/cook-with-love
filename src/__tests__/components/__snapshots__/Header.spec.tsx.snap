// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
  id="header"
>
  <div
    className="header__text"
  >
    <h1>
      <Trans
        t={
          [MockFunction] {
            "calls": Array [
              Array [
                "cook-with-love-header-subtitle",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": "cook-with-love-header-subtitle",
              },
            ],
          }
        }
      >
        cook-with-love-header-title
      </Trans>
    </h1>
    <h5
      className="header__text-subtitle"
    >
      cook-with-love-header-subtitle
    </h5>
  </div>
</header>
`;
